plugins/janus_videoroom.c

```diff
@@ -6366,11 +6366,14 @@ static void *janus_videoroom_handler(void *data) {
 				/* Also notify event handlers */
 				if(notify_events && gateway->events_is_enabled()) {
 					json_t *info = json_object();
+					json_t *token = json_object_get(root, "token");
 					json_object_set_new(info, "event", json_string("joined"));
 					json_object_set_new(info, "room", string_ids ? json_string(publisher->room->room_id_str) :
 					json_integer(publisher->room->room_id));
 					json_object_set_new(info, "id", string_ids ? json_string(user_id_str) : json_integer(user_id));
 					json_object_set_new(info, "private_id", json_integer(publisher->pvt_id));
+                    if (publisher->room->check_allowed)
+    					json_object_set_new(info, "token", token);
```

Comment:
Code style: no space between `if` and bracket.

I think something may go wrong with the Jansson refcount mechanism the way you're taking the object from one and adding it to another. Better to read the constant string and set that. On a related note, I'd only read `token` if the check succeeds, so something like:

```
if(publisher->room->check_allowed) {
	const char *token = json_string_value(json_object_get(root, "token"));
	json_object_set_new(info, "token", json_string(token));
}
```

Reply:
thx for having a look into this, i just updated the pr